### PR TITLE
bump crengine: rework char categorization and word selection

### PR DESCRIPTION
**Pull request by poire-z**
Includes https://github.com/koreader/crengine/pull/472 :
- LVString: fix char props, add lStr_isCJK() and lStr_isRTL()
- Rework is/prev/next-VisibleWord-Start/End()
- getCursorRect(): fix bottom point check

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1471)

### CODE REVIEW

File: cre.cpp [hunk 1]
```diff
@@ -1914,7 +1914,10 @@ static int getTextFromPositions(lua_State *L) {
 		// It's usually OK with multiple chars selection, even if one needs to go over the further
 		// 2nd half of the glyph to have it included in the selection.
 		// But it is less OK with an initial long-press on a glyph, where we get a 50% chance of
-		// having the next char selected. Try to handle this case better:
+		// having the next char selected. Try to handle this case better.
+		// It turns out this also happens when some CJK/symbol char is near a multi-alpha word,
```
Comment (Frenzie) [inline cre.cpp:1918]: multi-apha word?

Comment (poire-z) [reply cre.cpp:1918]: How you would name a word with more than one alphanumerical chars ? vs. CJK and symbols &copy; which are technically words in this context of text selection.

**Comment by poire-z** [comment]
@Frenzie : is there something similar to https://github.com/koreader/crengine/pull/471 (and the other stuff you fixed recently) to update in koreader-base ? We get here both:

```
Cloning into 'busted'...
fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
```

and

```
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/gcc-9/gcc-9-multilib_9.3.0-17ubuntu1~20.04_amd64.deb  404  Not Found [IP: 52.154.174.208 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
Error: Process completed with exit code 100.
```

**Comment by Frenzie** [comment]
I didn't fix anything in base yet but GHA can be safely ignored.

**Comment by Frenzie** [comment]
#1465 is a working fix, #1466 for some reason is not

File: cre.cpp (hunk 1 above)
Comment (Frenzie) [reply cre.cpp:1918]: Not complaining, just not sure what it meant.

Maybe multicharacter?

**Pull request merged**